pkg/controller/multiarchbuildconfig/multiarchbuildconfig.go

```diff
@@ -120,41 +113,43 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, logge
 	return nil
 }
 
-func (r *reconciler) handleMultiArchBuildConfig(ctx context.Context, mabc *v1.MultiArchBuildConfig) error {
+func (r *reconciler) handleMultiArchBuildConfig(ctx context.Context, mabc *v1.MultiArchBuildConfig) (resultErr error) {
+	var mutateFn func(mabcToMutate *v1.MultiArchBuildConfig)
+
+	defer func() {
```

Comment:
Yea it accepts a mutate function to make it more generic. Since you are handling each step, you can just update the mabc accordingly by constructing the changes that you need to make.

Reply:
So we can remove `UpdateMultiArchBuildConfig` entirely then?

Reply:
Why? What I mean is that you can just return the UpdateMultiArchBuildConfig error. No need to do the defer tricks to append the existing `resultErr`. Either way, If you check my other comments, you can just get rid of the whole defer and update the mabc on each of the method. No need to return the mutateFn to pass it to the defer operation.

Reply:
See https://github.com/openshift/ci-tools/pull/3676#discussion_r1381726683